Added github action to build and push to dockerhub

yarden-fishler-dev opened the pull request.

yarden-fishler-dev committed:
Added github action to build and push to dockerhub

yarden-fishler-dev committed:
Made the docker tag configurable and allowed for manual dispatch

yarden-fishler-dev committed:
Lint fixes

yarden-fishler-dev committed:
Changed it to run after the lint rather thn duplicate it

yarden-fishler-dev committed:
Lint fixes

yarden-fishler-dev committed:
Changed to github PAT rather than password

yarden-fishler-dev committed:
Tried to add branch input

RLRabinowitz commented on .github/workflows/build-and-deploy.yml:
```diff
@@ -0,0 +1,78 @@
+    workflows: ['Lint']
+    types: [completed]
+    branches: [main]
+  workflow_dispatch:
```
We'll also want the release to automatically do this. Using a release event or push tags event

yarden-fishler-dev replied:
2eb9262596ed5ca0cb69a8d8fb71cdbb2d861ab6

yarden-fishler-dev committed:
Add release and tag triggers to build workflow

- Add support for automatic releases via release events
- Add support for tag-based releases (v* tags)
- Update Docker tagging strategy for releases
- Add semver tag patterns for better version management
- Update job conditions to handle new trigger events
- Improve build summary with trigger information

Addresses PR review comment about automatic releases

yarden-fishler-dev committed:
Refactor build workflow to reuse existing lint workflow

- Remove main branch automatic trigger from build workflow
- Add lint job that calls existing ./.github/workflows/lint.yml
- Make build-and-push job depend on lint job passing
- Update lint.yml to support workflow_call trigger
- Eliminate code duplication between workflows
- Ensure code quality gate before any deployments

This creates a cleaner, DRY approach while maintaining the same functionality.

yarden-fishler-dev committed:
fix: correct Docker latest tag to use semantic versioning

Previously, the 'latest' tag was applied based on the default branch
(is_default_branch), which meant any build from main/master would get
the latest tag, including beta builds.

This change updates the tagging strategy to use semantic versioning
for the latest tag (type=semver,pattern=latest), ensuring that:

- 'latest' only applies to actual version releases (v1.2.3)
- Beta builds don't interfere with the latest stable release
- The latest tag truly represents the most recent semantic version

This follows Docker best practices where 'latest' should point to
the most recent stable release, not arbitrary builds from default branch.
